actions/user_actions.jsx

```diff
@@ -179,6 +179,21 @@ export async function loadNewGMIfNeeded(channelId) {
     checkPreference();
 }
 
+export function loadProfilesForGroupChannels(groupChannels) {
+    return (doDispatch, doGetState) => {
+        const state = doGetState();
+        const userIdsInChannels = Selectors.getUserIdsInChannels(state);
+
+        for (const {id} of groupChannels) {
+            const userIdsInGroupChannel = (userIdsInChannels[id] || new Set());
+
+            if (userIdsInGroupChannel.size === 0) {
+                doDispatch(UserActions.getProfilesInChannel(id, 0, Constants.MAX_USERS_IN_GM));
```

Comment:
You are potentially generating 50 requests, right?

Reply:
I'm thinking about a more general api endpoint, something like "getProfilesInChannels" and receive a list of channels, instead of just one channel.